apps/wallet/src/ui/app/pages/home/nft-transfer/useTransferKioskItem.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    ORIGINBYTE_KIOSK_MODULE,
+    ORIGINBYTE_KIOSK_OWNER_TOKEN,
+    useGetOwnedObjects,
+    useRpcClient,
+} from '@mysten/core';
+import { type SuiAddress, TransactionBlock } from '@mysten/sui.js';
+import { useMutation } from '@tanstack/react-query';
+
+import { useActiveAddress, useSigner } from '_src/ui/app/hooks';
+
+export function useTransferKioskItem({
+    objectId,
+    objectType,
+}: {
+    objectId: string;
+    objectType?: string;
+}) {
+    const rpc = useRpcClient();
+    const signer = useSigner();
+    const address = useActiveAddress();
+
+    const { data: kioskOwnerTokens } = useGetOwnedObjects(address, {
+        StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN,
+    });
+    const kioskIds = kioskOwnerTokens?.pages
+        .flatMap((page) => page.data)
+        .map(
+            (obj) =>
+                obj.data?.content &&
+                'fields' in obj.data.content &&
+                obj.data.content.fields.kiosk
+        );
+
+    return useMutation({
+        mutationFn: async (to: SuiAddress) => {
+            if (!to || !signer || !objectType) {
+                throw new Error('Missing data');
+            }
+
+            const tx = new TransactionBlock();
+
+            // fetch the kiosks for the active address
+            const ownedKiosks = await rpc.multiGetObjects({
+                ids: kioskIds!,
+                options: { showContent: true },
+            });
+
+            // find the kiosk id containing the object that we want to transfer
+            const kioskId = ownedKiosks.find(async (kiosk) => {
```

Comment:
This seems a bit expensive. Maybe we can pass the kioskId? If this is cached that is fine.

Reply:
yea, I'm planning to follow up to optimize this a bit more

Reply:
Yeah why not just use the owner? Shouldn't that chain be the kiosk?